detox/android/detox/src/test/java/com/wix/detox/DetoxActionHandlersSpec.kt

```diff
@@ -142,5 +142,82 @@ object DetoxActionHandlersSpec: Spek({
                         eq(messageId))
             }
         }
+
+        describe("Recording state actions") {
+            lateinit var instrumentsManager: DetoxInstrumentsManager
+
+            fun uut() = RecordingStateActionHandler(instrumentsManager, wsClient)
+
+            beforeEachTest {
+                instrumentsManager = mock()
+            }
+
+            it("should start recording with path") {
+                uut().handle("{\"recordingPath\":\"/MockPath\"}", messageId)
+                verify(instrumentsManager).startRecordingAtLocalPath(eq("/MockPath"))
+            }
+
+            it("should stop recording without path") {
+                uut().handle("{\"recordingPath\":null}", messageId)
+                verify(instrumentsManager).stopRecording()
+            }
+
+            it("should reply with a 'done' ACK on set state finish") {
+                uut().handle(params, messageId)
+                verify(wsClient).sendAction(eq("setRecordingStateDone"), any(), eq(messageId))
+            }
+        }
+
+        describe("Events actions") {
+            lateinit var instrumentsManager: DetoxInstrumentsManager
+
+            fun uut() = EventsActionsHandler(instrumentsManager, wsClient)
+
+            beforeEachTest {
+                instrumentsManager = mock()
+            }
+
+            describe("begin interval") {
+                val json = "{\"category\":\"MockCategory\",\"name\":\"MockName\",\"id\":\"MockId\",\"additionalInfo\":\"MockAdditionalInfo\",\"action\":\"begin\"}"
```

Comment:
this is impossible to read :) let's figure out a friendlier solution here. For example, a parser of a multiline JSON string. Or maybe a JSON builder, finally passing to the uut it's parsed (string) form...

Reply:
It's actually gets worse when this content is used in the call to `verify`. Need to come up with a way to predefine and reuse these strings...